src/tiler/healthCheck.js

```diff
@@ -0,0 +1,105 @@
+var Step = require('windshaft/node_modules/step');
+var Pg = require('windshaft/node_modules/cartodb-psql/node_modules/pg');
+var Redis = require('windshaft/node_modules/redis-mpool/node_modules/redis');
+
+/**
+ * Attempts to connect to a PostgreSQL database, providing the
+ * callback with a success or failure boolean.
+ * @param {string} host
+ * @param {string} port
+ * @param {string} user
+ * @param {string} password
+ * @param {string} database
+ * @param {function} callback
+ */
+function checkPostgres(host, port, user, password, database, callback) {
+  var conString = ['postgres://', user, ':', password, '@', host, '/', database].join('');
+  var client = new Pg.Client(conString);
+
+  client.connect(function(err) {
+    client.end();
+    callback(null, !err);
+  });
+}
+
+/**
+ * Attempts to connect to a Redis data store, providing the
+ * callback with a success or failure boolean.
+ * @param {string} host
+ * @param {string} port
+ * @param {function} callback
+ */
+function checkRedis(host, port, callback) {
+  var client = Redis.createClient(port, host);
+  var wrapper = function(status) {
+    client.quit();
+    callback(null, status);
+  };
+
+  client.on('connect', function() {
+    wrapper(true);
+  });
+
+  client.on('error', function() {
+    wrapper(false);
+  });
+}
+
+/**
+ * Makes use of the PostgreSQL and Redis checks above to return
+ * a JSON object representing the health of both data stores:
+ *
+ * {
+ *   "databases": [
+ *     {
+ *       "default": {
+ *         "ok": true
+ *       }
+ *    }
+ *   ],
+ *   "caches": [
+ *     {
+ *       "default": {
+ *         "ok": true
+ *       }
+ *     }
+ *   ]
+ * }
+ *
+ * @param {object} config - The configuration object passed to Windshaft.Server()
+ */
+module.exports = function createHealthCheckHandler(config) {
+  var pgConf = config.grainstore.datasource,
+
+    pgHost = pgConf.host,
+    pgPort = pgConf.port,
+    pgUser = pgConf.user,
+    pgPassword = pgConf.password,
+    pgDatabase = pgConf.dbname,
```

Comment:
The tiler health check fails for me because `dbName` field is missing from the config.

Error:

```
{ [error: database "vagrant" does not exist]
  name: 'error',
  length: 85,
  severity: 'FATAL',
  code: '3D000',
  detail: undefined,
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  file: 'postinit.c',
  line: '794',
  routine: 'InitPostgres' }
```

Config:

```
{ base_url: '/:tableId',
  base_url_notable: '/:tableId',
  grainstore:
   { datasource:
      { user: 'mmw',
        host: '33.33.34.30',
        port: '5432',
        password: 'mmw',
        geometry_field: 'geom',
        srid: 4326 },
     mapnik_version: '2.3.0',
     carto_env: { validation_data: [Object] },
     gc_prob: 0,
     cachedir: '/tmp/millstone' },
  renderCache: { ttl: 60000, statsInterval: 60000 },
  mapnik: { metatile: 4, bufferSize: 64 },
  redis:
   { host: '33.33.34.30',
     port: '6379',
     name: 'windshaft:server',
     max: 50,
     idleTimeoutMillis: 10000,
     reapIntervalMillis: 1000,
     unwatchOnRelease: true,
     log: false,
     slowPool: { log: false, elapsedThreshold: 25 } },
     ...
```